js/src/common/models/Discussion.ts

```diff
@@ -0,0 +1,146 @@
+import app from '../../common/app';
+import Model from '../Model';
+import computed from '../utils/computed';
+import ItemList from '../utils/ItemList';
+import Badge from '../components/Badge';
+import Mithril from 'mithril';
+import Post from './Post';
+import User from './User';
+
+export default class Discussion extends Model {
+  title() {
+    return Model.attribute<string>('title').call(this);
+  }
```

Comment:
wouldn't it be more friendly for attributes/relationships to write instead:
```suggestion
  title = Model.attribute<string>('title');
```

Reply:
@SychO9 This breaks monkey patching I believe. Class properties are not in `.prototype` and thus cannot be extended easily (if at all (?))

From Clark's PR
> The difference between `Object.assign`/`mixin` and the proposed solution is that the attributes are no longer part of the model `prototype` and as such cannot be easily overridden by extensions. However I don't think any extension ever attempted that and I'm not sure why the "mixin" style was used at all in the past.

More @ https://github.com/flarum/core/pull/2411#issuecomment-847427695

Reply:
ah shame :(, the easier the code is the easier it is for new extension devs to get into it. But being able to monkey patch these is important as well.

Reply:
Personally, I don't think model attributes should really be monkey patched (that's a VERY unsafe approach to development), just that for now we need to maintain BC.

Reply:
It will create a difference between ext and core attributes, as we extend Models with `prototype`, and once we switch over to class properties, those would be separate from the prototype. Don't think this would cause issues, but it might be confusing. 

And yeah, think I'm for changing it in a breaking release as it's very ugly and repetitive.

Reply:
I'm hoping that by the time 2.0 rolls around, we will have proper JS extenders for this.